### PR TITLE
Improve reader, fix some tests

### PR DESCRIPTION
**Pull request by azaroth42**
Reader now  works with add_attribute_assignment and conceptual parts.
Also creates instances of vocab "extension" classes,  rather than vanilla classes  from the  model with the right classifications.
This is important for boundary checking, in case the test looks for the vocab class, rather than the classification.

Also fix some edge cases revealed by testing the reader on all the examples in linked art site, and add tests.

### CODE REVIEW

File: cromulent/vocab.py [hunk 1]
```diff
@@ -582,7 +582,7 @@ def rights_getter(self, what):
 		elif what == "part_of":
 			return self.c_part_of
 		else:
-			getattr(self, what)
+			object.__getattr__(self, what)
```
Comment (kasei) [inline cromulent/vocab.py]: Why is this change needed?

Comment (azaroth42) [reply cromulent/vocab.py]: It works as expected in python2, but in python3 it fails with an infinite recursion error, when the object is being constructed from the reader. I don't know why exactly!

File: tests/test_reader.py [hunk 2]
```diff
@@ -41,3 +43,27 @@ def test_read(self):
...
+		  "assigned_to": {
+		    "id": "https://linked.art/example/object/12", 
+		    "type": "HumanMadeObject", 
+		    "_label": "Real Painting Name"
+		  }
+		}
+		"""
+		d = self.reader.read(data)
+		
+
```
Comment (kasei) [inline tests/test_reader.py]: This test doesn't seem to have any assertions. Was it meant to be filled in?

Comment (azaroth42) [reply tests/test_reader.py]: Oh, heh.  I just had it do the read and fixed errors... it probably should have an assert or two as well.

Comment (kasei) [reply tests/test_reader.py]: OK. Modulo this missing test, everything looks good.

**Comment by azaroth42** [comment]
@bluebinary This should now work to read in data into the internal object form crom uses.  Let me know if there are any issues.
